hittingEndPoint.vb

```diff
@@ -0,0 +1,176 @@
+﻿Imports System.Windows.Forms
+Imports System.Net
+Imports System.IO
+Imports System.Text
+Module hittingEndPoint
+    
+    'TODO: To understand "getAuthorizationToken()" function see this:
+    'https://www.example-code.com/vbnet/hmrc_oauth2_access_token.asp 
+    Private Function getAuthorizationToken(ByRef userClientId As String , ByRef userClientSecretId As String) As String
+        Dim glob As New Chilkat.Global
+        Dim success As Boolean = glob.UnlockBundle("Anything for 30-day trial")
+        If (success <> True) Then
+            MessageBox.Show(glob.LastErrorText)
+            Return ""
+        End If
+        Dim oauth2 As New Chilkat.OAuth2
+        oauth2.ListenPort = 55568
+        oauth2.AuthorizationEndpoint = "https://accounts.google.com/o/oauth2/auth"
+        oauth2.TokenEndpoint = "https://oauth2.googleapis.com/token"
+        'TODO: set your "clientId" and clientSecretId".
+        oauth2.ClientId = userClientId
+        oauth2.ClientSecret = userClientSecretId
+        oauth2.CodeChallenge = True
+        oauth2.CodeChallengeMethod = "S256"
+        oauth2.Scope = "https://www.googleapis.com/auth/spreadsheets"
+        Dim url As String = oauth2.StartAuth()
+        If (oauth2.LastMethodSuccess <> True) Then
+            MessageBox.Show(oauth2.LastErrorText)
+            Return ""
+        End If
+        System.Diagnostics.Process.Start(url)
+        Dim numMsWaited As Integer = 0
+        While (numMsWaited < 30000) And (oauth2.AuthFlowState < 3)
+            oauth2.SleepMs(100)
+            numMsWaited = numMsWaited + 100
+        End While
+
+        If (oauth2.AuthFlowState < 3) Then
+            oauth2.Cancel()
+            MessageBox.Show("No response from the browser!")
+            Return ""
+        End If
+        If (oauth2.AuthFlowState = 5) Then
+            MessageBox.Show("OAuth2 failed to complete.")
+            MessageBox.Show(oauth2.FailureInfo)
+            Return ""
+        End If
+        If (oauth2.AuthFlowState = 4) Then
+            MessageBox.Show("OAuth2 authorization was denied.")
+            MessageBox.Show(oauth2.AccessTokenResponse)
+            Return ""
+        End If
+        If (oauth2.AuthFlowState <> 3) Then
+            MessageBox.Show("Unexpected AuthFlowState:" & oauth2.AuthFlowState)
+            Return ""
+        End If
+        Return oauth2.AccessToken
+    End Function
+
+    Public Function callSheetsAPI(ByRef fileId As String, ByRef userClientId As String , ByRef userClientSecretId As String) As List(Of String)
+        Dim lines As List(Of String) = Nothing
+        'Calls for Authorization Token.
+        Dim _bearerToken As String = getAuthorizationToken(userClientId,userClientSecretId)
+        If _bearerToken <> "" Then
+            Dim myUri As New Uri("https://docs.google.com/spreadsheets/vbaprocessfile?fid=" + fileId)

```

Comment:
Create string constant for https://docs.google.com/spreadsheets/vbaprocessfile?fid=

Reply:
It is constant because here it is pass by reference and from where i called it there "fileId" is defined as constant. Because there is no direct way to define argument as constant.